pandas/indexes/numeric.py

```diff
@@ -176,26 +188,81 @@ def _assert_safe_casting(cls, data, subarr):
 Int64Index._add_numeric_methods()
 Int64Index._add_logical_methods()
 
+_uint64_descr_args = dict(
+    klass='UInt64Index',
+    dtype='uint64',
+    extra=''
+)
 
-class Float64Index(NumericIndex):
-    """
-    Immutable ndarray implementing an ordered, sliceable set. The basic object
-    storing axis labels for all pandas objects. Float64Index is a special case
-    of `Index` with purely floating point labels.
 
-    Parameters
-    ----------
-    data : array-like (1-dimensional)
-    dtype : NumPy dtype (default: object)
-    copy : bool
-        Make a copy of input ndarray
-    name : object
-        Name to be stored in the index
+class UInt64Index(NumericIndex):
+    __doc__ = _num_index_shared_docs['class_descr'] % _uint64_descr_args
 
-    Notes
-    -----
-    An Float64Index instance can **only** contain hashable objects
-    """
+    _typ = 'uint64index'
+    _arrmap = _algos.arrmap_uint64
+    _left_indexer_unique = _join.left_join_indexer_unique_uint64
```

Comment:
Perhaps...though most of the boilerplate you are seeing is from `UInt64Index` I think (I copied and pasted `uint64`).  Perhaps we can do that as a follow-up once `UInt64Index` is added?